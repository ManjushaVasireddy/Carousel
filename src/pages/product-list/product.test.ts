import "jest";
import { VueConstructor } from "vue";
import { shallowMount, createLocalVue } from "@vue/test-utils";

import Product from "./product.vue";
import mock from "./product.mock";

const localVue: VueConstructor<Product> = createLocalVue();

describe("Product.vue", () => {
    const wrapper: any = shallowMount(Product, {
        localVue,
        propsData: { item: mock.Product, cardSelectedCall: mock.cardSelectedCall }
    });

    it("shows the product", () => {
        expect(wrapper.find(".product .price").text().toBeDefined);
    });

    it("matches snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
