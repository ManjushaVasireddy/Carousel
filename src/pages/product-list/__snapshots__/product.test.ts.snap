// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product.vue matches snapshot 1`] = `
<section class="product">
  <card-stub imagelink="https://assets.weimgs.com/weimgs/ab/images/wcm/products/202122/0113/organic-washed-cotton-percale-lightweight-quilt-shams-1-m.jpg" altlink="https://www.westelm.com/products/washed-cotton-percale-lightweight-quilt-shams-b3603/" headertext="Organic Washed Cotton Percale Lightweight Quilt &amp;amp; Shams">
    <!---->
    <p class="price product-sale">
      Sale: $34.5 - $269
    </p>
  </card-stub>
</section>
`;
